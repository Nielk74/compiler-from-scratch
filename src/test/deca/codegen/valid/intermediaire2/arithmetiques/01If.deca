// Description:
//    Programme minimaliste réalisant des comparaisons dans des if entre une variable int et le résultat d'opéartions arithmétiques.
//
// Resultats:
//    01if.expected
//
// Historique:
//    cree le 08/06/2022

{
    int x = 0;
    if (x < 1 + 1) {
        println(x, " < 1 + 1");
    } else {
        println(x, " >= 1 + 1");
    }

    if (x < 1 - 1) {
        println(x, " < 1 - 1");
    } else {
        println(x, " >= 1 - 1");
    }

    if (x < 1 * 1) {
        println(x, " < 1 * 1");
    } else {
        println(x, " >= 1 * 1");
    }

    if (x == 1 / 1) {
        println(x, " == 1 / 1");
    } else {
        println(x, " != 1 / 1");
    }

    x = 5;
    if (x == 12 % 7) {
        println(x, " == 12 % 7");
    } else {
        println(x, " != 12 % 7");
    }

    if (12 % 7 == x) {
        println("12 % 7 == ", x);
    } else {
        println("12 % 7 != ", x);
    }
}