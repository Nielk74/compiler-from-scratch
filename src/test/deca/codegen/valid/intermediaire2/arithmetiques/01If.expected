0 < 1 + 1
0 >= 1 - 1
0 < 1 * 1
0 != 1 / 1
5 == 12 % 7
12 % 7 == 5
